### PR TITLE
[WSSE] - Using a PSR6 cache instead of file cache

### PR DESCRIPTION
**Pull request by Nyholm**
I rewrote the WSSE example to use a PSR-6 cache instead. I did this because Symfony got a fancy new CacheComponent now and because using file cache will eventually fill up your inodes. (I just removed 2 million inodes from my server.)

Note: Im not sure about the service name for a cache pool. Is it just `cache`?

### CODE REVIEW

File: cookbook/security/custom_authentication_provider.rst [hunk 1]
```diff
@@ -127,7 +127,7 @@ set an authenticated token in the token storage if successful.
 
         public function __construct(TokenStorageInterface $tokenStorage, AuthenticationManagerInterface $authenticationManager)
         {
-            $this->tokenStorage = $tokenStorage;
+            $this->tokenStorage          = $tokenStorage;
```
Comment (xabbuh) [inline cookbook/security/custom_authentication_provider.rst]: Please revert this change (we do not align assignments in Symfony).

File: cookbook/security/custom_authentication_provider.rst [hunk 2]
```diff
@@ -218,12 +219,12 @@ the ``PasswordDigest`` header value matches with the user's password.
     class WsseProvider implements AuthenticationProviderInterface
     {
         private $userProvider;
-        private $cacheDir;
+        private $cachePool;
 
-        public function __construct(UserProviderInterface $userProvider, $cacheDir)
+        public function __construct(UserProviderInterface $userProvider, CacheItemPoolInterface $cachePool)
```
Comment (xabbuh) [inline cookbook/security/custom_authentication_provider.rst]: Imo we should type hint against the [AdapterInterface](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/AdapterInterface.php) from the Cache component. This way we can inject cache adapters defined through the `framework.cache` config (see below).

File: cookbook/security/custom_authentication_provider.rst [hunk 3]
```diff
@@ -411,7 +411,7 @@ to service ids that do not exist yet: ``wsse.security.authentication.provider``
                 class: AppBundle\Security\Authentication\Provider\WsseProvider
                 arguments:
                     - '' # User Provider
-                    - '%kernel.cache_dir%/security/nonces'
+                    - '@cache'
```
Comment (xabbuh) [inline cookbook/security/custom_authentication_provider.rst]: We should inject the `cache.app` service here imo.

**Comment by xabbuh** [comment]
I basically like the idea to make use of the new Cache component in this example. Thanks for the heads up @Nyholm! I just left some minor comments about changes we should imo make to better fit with how the PSR-6 cache system is integrated in Symfony.

**Comment by xabbuh** [comment]
@weaverryan @WouterJ By the way, we need to pay attention to merge this into the `3.1` branch.

File: cookbook/security/custom_authentication_provider.rst (hunk 1 above)
Comment (Nyholm) [inline cookbook/security/custom_authentication_provider.rst]: Sorry. I looked at the other constructor on this page where you do align it. I'll revert it. Thank you.

File: cookbook/security/custom_authentication_provider.rst (hunk 2 above)
Comment (Nyholm) [inline cookbook/security/custom_authentication_provider.rst]: No. We should not type hint against AdapterInterface. We want to use the PSR. Becasue we do not want to force the user to use Symfonys cache component. Let's use the strength of the PSRs. :)

**Comment by Nyholm** [comment]
Thank you for your comments. I'll update the pr tomorrow.

File: cookbook/security/custom_authentication_provider.rst (hunk 2 above)
Comment (WouterJ) [inline cookbook/security/custom_authentication_provider.rst]: AdapterInterface extends the PSR ones, so no problem here and indeed: Let's be as generic as possible

**Comment by Nyholm** [comment]
I've just pushed some changes to the PR.

File: cookbook/security/custom_authentication_provider.rst (hunk 2 above)
Comment (xabbuh) [inline cookbook/security/custom_authentication_provider.rst]: You are right. There's no need to require the more specialised interface here.

**Comment by xabbuh** [comment]
👍 Great work @Nyholm!

**Comment by WouterJ** [comment]
Thanks Tobias, this looks very nice!

**Comment by Nyholm** [comment]
Thank you for merging.
